documents/model.go

```diff
@@ -754,3 +755,154 @@ func isNFTInRole(role *coredocumentpb.Role, registry common.Address, tokenID []b
 
 	return false
 }
+
+// assembleAccessToken assembles a Read Access Token from the payload received
+func assembleAccessToken(dm *CoreDocumentModel, id identity.IDConfig, payload documentpb.AccessTokenParams) (*coredocumentpb.AccessToken, error) {
+	tokenIdentifier := utils.RandomSlice(32)
+	granterID := id.ID[:]
+	roleID := byte(len(dm.Document.Roles))
+	grantee, err := hexutil.Decode(payload.GetGrantee())
+	if err != nil {
+		return nil, err
+	}
+	docID, err := hexutil.Decode(payload.GetDocumentIdentifier())
+	if err != nil {
+		return nil, err
+	}
+	// assemble access token message to be signed
+	tm := append(tokenIdentifier, granterID...)
+	tm = append(tm, grantee...)
+	tm = append(tm, roleID)
+	tm = append(tm, docID...)
+
+	// fetch key pair from identity
+	privateKey := id.Keys[identity.KeyPurposeEthMsgAuth].PrivateKey
+	pubKey := id.Keys[identity.KeyPurposeEthMsgAuth].PublicKey
+
+	// sign the assembled access token message, return signature
+	sig, err := crypto.SignMessage(privateKey, tm, "CurveSecp256K1", true)
```

Comment:
it would be nice to expose the curves as constants in the crypto package and then refer to it instead of adding a string here, @vimukthi-git you initially refactored the crypto package. What do you think?

Reply:
This was on my mind as well, it makes sense to add constants.

Reply:
Actually they are already public, you can use: crypto.CurveSecp256K1
```suggestion
	sig, err := crypto.SignMessage(privateKey, tm, crypto.CurveSecp256K1, true)
```